go/obscuronode/host/host.go

```diff
@@ -485,12 +498,21 @@
 	for _, tx := range b.Transactions() {
 		t := a.txHandler.UnPackTx(tx)
 		if t != nil && t.TxType == obscurocommon.RequestSecretTx {
-			txData := &obscurocommon.L1TxData{
-				TxType:      obscurocommon.StoreSecretTx,
-				Secret:      a.Enclave.FetchSecret(t.Attestation),
-				Attestation: t.Attestation,
+			att, err := nodecommon.DecodeAttestation(t.Attestation)
+			if err != nil {
+				log.Log("Failed to decode attestation")
+			}
+			secret, err := a.Enclave.FetchSecret(*att)
```

Comment:
I think if there was a continue here then the else keyword could be removed

Reply:
Yeah that didn't smell right at the time, thanks